### ISSUE TITLE
Предлагаю добавить команду для inline вывода

### ISSUE THREAD

**Issue by MioVisman** [artoodetoo/dirk#1]
Привет!

Предлагаю добавить две команды `@inline` и `@endinline` для удаления в шаблоне между ними (+ перевод строки до `@inline`) всех пробельных символов из начала строк + переводы строк.

**Для чего это нужно:** вывод inline элементов на страницу без пробелов. Можно конечно и костыли использовать как в самом шаблоне `<!-- -->`, так и в css, но это муторно.

Пример:
Скомпилированный шаблон до:

    <?php if($cur['moderators']): ?>
                  <p class="f-modlist">
                    <span><?= __('Moderated by') ?></span>
    <?php foreach($cur['moderators'] as $mod): ?>
    <?php if(is_string($mod)): ?>
                    <?= htmlspecialchars($mod, ENT_HTML5 | ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8') ?>
    <?php else: ?>
                    <a href="<?= $mod[0] ?>"><?= htmlspecialchars($mod[1], ENT_HTML5 | ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8') ?></a>
    <?php endif; ?>
    <?php endforeach; ?>
                  </p>
    <?php endif; ?>

В оригинале после `<p class="f-modlist">` и `</p>` поставили команды  `@inline` и `@endinline`. Получили:

    <?php if($cur['moderators']): ?>
                  <p class="f-modlist"><span><?= __('Moderated by') ?></span><?php foreach($cur['moderators'] as $mod): ?><?php if(is_string($mod)): ?><?= htmlspecialchars($mod, ENT_HTML5 | ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8') ?><?php else: ?><a href="<?= $mod[0] ?>"><?= htmlspecialchars($mod[1], ENT_HTML5 | ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8') ?></a><?php endif; ?><?php endforeach; ?></p>
    <?php endif; ?>

**Comment by MioVisman** [comment]
Хотя возможно стоит просто 4й `compiler` добавить, который будет комментарии html проверять. Если встретил `<!--inline-->` и `<!--endinline-->`, то преобразовывает скомпилированный код.

**Comment by MioVisman** [comment]
Для себя такой метод наваял

    /**
     * Трансформация скомпилированного шаблона
     * @param string $value
     * @return string
     */
    protected function compileTransformations($value)
    {
        if (strpos($value, '') === false) {
            return $value;
        }
        $value = preg_replace_callback(
            '%([^<]*(?:<(?!!--endinline-->)[^<]*)*+)(?:)?%',
            function ($matches) {
                return preg_replace('%\h*\R\s*%', '', $matches[1]);
            },
            $value);
        return $value;
    }

**Comment by artoodetoo** [comment]
Цель какая? Решать проблему со стилями лучше самими стилями. Иначе это какие-то костыли.

Посмотри вот это, я угадал с целью?

- https://css-tricks.com/fighting-the-space-between-inline-block-elements/
- http://stackoverflow.com/questions/5078239/how-to-remove-the-space-between-inline-block-elements

**Comment by MioVisman** [comment]
Я про это и писал в стартовом сообщении :) Для себя решил добавлением compileTransformations(), может кому пригодится (код выше в моем сообщении).

**Comment by artoodetoo** [comment]
Конечно, надеюсь оно пригодится. Чем больше примеров использования и персональной настройки, тем лучше для будущих фанатов :)
Спасибо тебе за поддержку!
